server/sonar-server/src/main/java/org/sonar/server/user/UserUpdater.java

```diff
@@ -65,13 +63,15 @@
   private final DbClient dbClient;
   private final UserIndexer userIndexer;
   private final System2 system2;
+  private final SecurityRealmFactory realmFactory;
 
-  public UserUpdater(NewUserNotifier newUserNotifier, Settings settings, DbClient dbClient, UserIndexer userIndexer, System2 system2) {
+  public UserUpdater(NewUserNotifier newUserNotifier, Settings settings, DbClient dbClient, UserIndexer userIndexer, System2 system2, SecurityRealmFactory realmFactory) {
```

Comment:
As its name implies, it provides access to the SecurityRealm implementation used by the platform, if any (LDAP, Crowd, etc.)